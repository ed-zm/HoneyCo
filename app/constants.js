export const HoneyCoYellow = '#ffd34c'
export const HoneyCoDarkYellow = '#db9f33'
export const HoneyCoRed = '#bf4931'
export const HoneyCoBlue = '#4DA29F'
export const HoneyCoBlueRGBA = (transparency) => transparency ? `rgba(77, 162, 159, ${transparency})` : `rgba(77, 162, 159, 0)`
export const HoneyCoDarkBlue = '#165d63'
export const HoneyCoLightGray = '#e4e4e4'
export const HoneyCoDarkGray = '#5e5e5e'
