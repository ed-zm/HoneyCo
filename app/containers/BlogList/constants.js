/*
 *
 * BlogList constants
 *
 */

export const BLOG_LIST_FETCH_REQUESTED = 'app/BlogList/BLOG_LIST_FETCH_REQUESTED'
export const BLOG_LIST_FETCH_SUCCEEDED = 'app/BlogList/BLOG_LIST_FETCH_SUCCEEDED'
export const BLOG_LIST_FETCH_FAILED = 'app/BlogList/BLOG_LIST_FETCH_FAILED'
