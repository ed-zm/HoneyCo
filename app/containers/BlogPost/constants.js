/*
 *
 * BlogPost constants
 *
 */

export const BLOG_POST_FETCH_REQUESTED = 'app/BlogPost/BLOG_POST_FETCH_REQUESTED'
export const BLOG_POST_FETCH_SUCCEEDED = 'app/BlogPost/BLOG_POST_FETCH_SUCCEEDED'
export const BLOG_POST_FETCH_FAILED = 'app/BlogPost/BLOG_POST_FETCH_FAILED'
