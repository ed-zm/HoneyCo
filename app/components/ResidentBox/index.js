import React from 'react'
import { Col } from 'reactstrap'
import styled from 'styled-components'
import { HoneyCoLightGray, HoneyCoDarkGray, HoneyCoYellow } from '../../constants'
const StyledCol = styled(Col)`
  background: url(${(props) => props.background});
  background-color: ${(props) => props.background};
  height: 600px;
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  padding: 2em;
  position: relative;
`
const StyledTitleText = styled.span`
  font-size: 40px;
  display: block;
  color: ${(props) => props.textColor};
  text-align: left;
  margin-bottom: 1em;
`
const StyledParagraph = styled.span`
  display: block;
  text-align: left;
  color: ${HoneyCoDarkGray};
  font-size: 22px;
  margin-bottom: 1em;
`

const StyledButton = styled.button`
  font-family: Oswald;
  font-size: 26px;
  background-color: ${HoneyCoYellow};
  padding: .2em;
  border-radius: 5px;
  border: none;
  padding: .3em 1.5em;
  text-align: center;
  margin-top: 10px;
`
const StyledList = styled.span`
  display: block
`

const ResidentBox = (props) =>
    <StyledCol xl={6} lg={6} md={12} sm={12} xs={12} background={props.background ? props.image : HoneyCoLightGray} color={props.color ? HoneyCoLightGray : ''}>
      <StyledTitleText textColor={props.textColor}>
        {props.text ? props.text.title : ''}
      </StyledTitleText>
      <StyledParagraph>
        {props.text ? props.text.firstParagraph : ''}
      </StyledParagraph>
      <StyledParagraph>
        {props.text ? props.text.secondParagraph : ''}
      </StyledParagraph>
      <StyledParagraph>
        {props.text ? props.text.thirdParagraph : ''}
      </StyledParagraph>
      <StyledParagraph>
        {props.text.list ? props.text.list.map((item) =>
          <StyledList>- {item}</StyledList>
        )
        : ''
        }
      </StyledParagraph>
      {props.text
      ? <StyledButton>{props.text.button}</StyledButton>
      : ''
      }
    </StyledCol>
export default ResidentBox
