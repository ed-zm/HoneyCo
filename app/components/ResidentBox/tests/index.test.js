import React from 'react'
import { shallow } from 'enzyme'
import ResidentBox from '../index'
import { HoneyCoBlue } from '../../../constants'

describe('<ResidentBox />', () => {
  it('It should render the ResidentBox Component', () => {
    const wrapper = shallow(<ResidentBox color={false} background={false} text={false} textColor={HoneyCoBlue}/>)
    expect(wrapper).toMatchSnapshot()
  })
})
