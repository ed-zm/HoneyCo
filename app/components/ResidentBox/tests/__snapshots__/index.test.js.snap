// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResidentBox /> It should render the ResidentBox Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(Ot)
    background="#e4e4e4"
    color=""
    lg={6}
    md={12}
    sm={12}
    xl={6}
    xs={12}
>
    <styled.span
        textColor="#4DA29F"
    />
    <styled.span />
    <styled.span />
    <styled.span />
    <styled.span />
    
</Styled(Ot)>,
  "nodes": Array [
    <Styled(Ot)
      background="#e4e4e4"
      color=""
      lg={6}
      md={12}
      sm={12}
      xl={6}
      xs={12}
>
      <styled.span
            textColor="#4DA29F"
      />
      <styled.span />
      <styled.span />
      <styled.span />
      <styled.span />
      
</Styled(Ot)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ResidentBox
        background={false}
        color={false}
        text={false}
        textColor="#4DA29F"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "background": false,
          "color": false,
          "text": false,
          "textColor": "#4DA29F",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(Ot)
          background="#e4e4e4"
          color=""
          lg={6}
          md={12}
          sm={12}
          xl={6}
          xs={12}
>
          <styled.span
                    textColor="#4DA29F"
          />
          <styled.span />
          <styled.span />
          <styled.span />
          <styled.span />
          
</Styled(Ot)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(Ot)
          background="#e4e4e4"
          color=""
          lg={6}
          md={12}
          sm={12}
          xl={6}
          xs={12}
>
          <styled.span
                    textColor="#4DA29F"
          />
          <styled.span />
          <styled.span />
          <styled.span />
          <styled.span />
          
</Styled(Ot)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ResidentBox
    background={false}
    color={false}
    text={false}
    textColor="#4DA29F"
/>,
}
`;
